.github/workflows/publish-snapshots.yml

```diff
@@ -3,7 +3,8 @@ name: Publish snapshots to maven
 on:
   workflow_dispatch:
   schedule:
-    - cron: 0 */12 * * *
+  push:
+    - '1.x'
```

Comment:
Need to add a `ref:` to `actions/checkout` to check out the correct branch.

Reply:
Same reason: https://github.com/opensearch-project/opensearch-sdk-java/pull/472#discussion_r1112309970?

Reply:
OK, I think. I'm still not certain which workflow will run on a push to 1.x.... docs seem to imply the default branch workflow will execute in both cases.